test/strscan/test_stringscanner.rb

```diff
@@ -944,6 +944,60 @@ def test_named_captures
     assert_equal({}, scan.named_captures)
     assert_equal(9, scan.match?(/(?<f>foo)(?<r>bar)(?<z>baz)/))
     assert_equal({"f" => "foo", "r" => "bar", "z" => "baz"}, scan.named_captures)
+
+    # test that named_captures has been reset
+    assert_equal(9, scan.match?("foobarbaz"))
+    assert_equal({}, scan.named_captures)
+
+    assert_equal(9, scan.match?(/(?<f>foo)(?<r>bar)(?<z>baz)/))
+    assert_equal({"f" => "foo", "r" => "bar", "z" => "baz"}, scan.named_captures)
+    scan.unscan
+    assert_equal({}, scan.named_captures)
```

Comment:
Could you move this to `test_unscan`? (Could you move other tests too?)

Reply:
OK.
I see.